Allow removing permute pairs in addition to transpose pairs

mcremon-meta opened the pull request:
Summary:
As titled. Gets us 27% better cycles on Activity Classification (at opt level 3).

Can be improved further, task is T222295719

Reviewed By: zonglinpeng

Differential Revision: D73619452

ThomasJannaud commented on backends/cadence/aot/fuse_ops.py:
```diff
@@ -907,42 +907,28 @@ def can_fuse_for_chain(
-            get_dims(producer),
-            get_dims(consumer),
-        )
+        return non_unit_dims == non_unit_dims_permuted
```
it seems that we are checking that the combination of 2 permutations is an identity based on the start and final size of the shapes, instead of on the permutation itself
(so if the input shape is [4 , 4, 4, 4] then this will always trigger)

ThomasJannaud committed:
Allow removing permute pairs in addition to transpose pairs (#10501)

Summary:

Pull Request resolved:

As titled. Gets us 27% better cycles on Activity Classification (at opt level 3).

Can be improved further (when fused permutations are not an identity), task is T222295719

Differential Revision: D73619452

ThomasJannaud commented:
closed in favor of #10566
